source/agora/flash/Channel.d

```diff
@@ -1580,6 +1606,8 @@ LOuter: while (1)
                 this.onSettleTxExternalized(tx);
             }
         }
+
+        this.checkPublishSettlement();
```

Comment:
Does this mean that we would wait for the next block to broadcast the settlement once we detect the update TX in a block?

Reply:
Ah you mean we shouldn't wait +1 blocks, but try to publish it right away? Let me check if it's doing that and fix it.

Reply:
> Does this mean that we would wait for the next block to broadcast the settlement once we detect the update TX in a block?

I think we do need to wait for the first block it's unlocked at because otherwise the tx would be rejected when it tries to be added to the pool. It won't be able to attach to the Output of the update tx as the script execution will fail.

Reply:
We should really fix the pool to accept future TXs. But in the meantime, yes.